creator-node/src/components/replicaSet/URSMRegistrationComponentService.js

```diff
@@ -0,0 +1,174 @@
+    throw new ErrorBadRequest(
+      `No-op - UserReplicaSetManager record for node with spID ${spID} already matches L1 ServiceProviderFactory record`
+    )
+  }
+
+  /**
+   * Confirm request was signed by delegate owner wallet registered on L1 for spID, given request signature artifacts
+   */
+  let requesterWalletRecoveryObj = { spID, timestamp: reqTimestamp }
+  let recoveredDelegateOwnerWallet = (recoverWallet(requesterWalletRecoveryObj, reqSignature)).toLowerCase()
+  if (delegateOwnerWalletFromSPFactory !== recoveredDelegateOwnerWallet) {
+    throw new ErrorBadRequest(
+      'Request for signature must be signed by delegate owner wallet registered on L1 for spID'
+    )
+  }
+
+  /**
+   * Confirm service provider is within valid stake bounds on L1 ServiceProviderFactory
+   */
+  const spDetailsFromSPFactory = await audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderDetails(
+    ownerWalletFromSPFactory
+  )
+  if (!spDetailsFromSPFactory.validBounds) {
+    throw new ErrorBadRequest('ServiceProvider for given spID is not within valid bounds on L1 ServiceProviderFactory')
+  }
+
+  /**
+   * Request node's up-to-date health info
+   *  - uses endpoint registered on L1 for spID
+   *  - passes `randomBytesToSign` string in request to check that response was signed for provided data
+   */
+  const randomBytesToSign = (await randomBytes(18)).toString()
+  let nodeHealthCheckResp = await axios({
+    baseURL: nodeEndpointFromSPFactory,
+    url: '/health_check',
+    method: 'get',
+    timeout: 1000,
+    params: {
+      randomBytesToSign
+    }
+  })
+  nodeHealthCheckResp = parseCNodeResponse(
+    nodeHealthCheckResp,
+    ['healthy', 'creatorNodeEndpoint', 'spID', 'spOwnerWallet', 'randomBytesToSign']
+  )
+
+  /**
+   * Confirm health check returns healthy and response data matches on-chain data
+   */
+  if (
+    !(nodeHealthCheckResp.healthy) ||
+    (nodeHealthCheckResp.creatorNodeEndpoint !== nodeEndpointFromSPFactory) ||
+    (nodeHealthCheckResp.spID !== spID) ||
+    ((nodeHealthCheckResp.spOwnerWallet).toLowerCase() !== ownerWalletFromSPFactory.toLowerCase())
+  ) {
+    throw new ErrorServerError(
+      `Content node health check response from endpoint ${nodeEndpointFromSPFactory} indicates unhealthy or misconfigured`
+    )
+  }
+
+  /**
+   * Confirm health check response was signed by delegate owner wallet registered on L1
+   *    for spID and includes `randomBytesToSign`
+   */
+  let {
+    timestamp: respTimestamp,
+    signature: respSignature,
+    ...responseData
+  } = nodeHealthCheckResp.rawResponse
+  if (signatureHasExpired(respTimestamp)) {
+    throw new ErrorBadRequest('Health check response signature has expired')
+  }
+  const responderWalletRecoveryObj = { ...responseData, randomBytesToSign, timestamp: respTimestamp }
+  recoveredDelegateOwnerWallet = (recoverWallet(responderWalletRecoveryObj, respSignature)).toLowerCase()
```

Comment:
lets set this to a new variable? this is a complicated flow, i'd advocate for not mutating stuff here

Reply:
fair, fixed